pcsx2/DEV9/AdapterUtils.cpp

```diff
@@ -244,30 +246,66 @@ std::optional<MAC_Address> AdapterUtils::GetAdapterMAC(Adapter* adapter)
 
 	return std::nullopt;
 }
-#elif defined(__POSIX__)
-#ifdef __linux__
+#else
 std::optional<MAC_Address> AdapterUtils::GetAdapterMAC(Adapter* adapter)
 {
-	struct ifreq ifr;
-	strcpy(ifr.ifr_name, adapter->ifa_name);
-
-	int fd = socket(AF_INET, SOCK_DGRAM, 0);
-	int ret = ioctl(fd, SIOCGIFHWADDR, &ifr);
-	close(fd);
+	MAC_Address macAddr = {0};
+#if defined(AF_LINK)
+	struct ifaddrs *adapterInfo, *po;
+	int error = getifaddrs(&adapterInfo);
+	if (error)
+	{
+		Console.Error("DEV9: Failed to get adapter information %s", error);
+		return std::nullopt;
+	}
+	for (po = adapterInfo; po != NULL; po = po->ifa_next)
+	{
+		if (strcasecmp(po->ifa_name, adapter->ifa_name))
```

Comment:
Reiterating an earlier question
Is there a reason you are using `strcasecmp` (for case insensitive comparison) rather than `strcmp`?

I ask as there is similar code elsewhere in this file using `strcmp`, and I want to know if that needs to be `strcasecmp`

Reply:
To be honest, I’m not sure, I don’t know if you can have the same name but with different case config. I was searching on the internet and I saw examples using case insensitivity.
I think it depends on OS case sensitive config.
I can remove if you think is the more appropriate

Reply:
I have removed it and used the common `strcmp` method.
Thanks!